flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/calcite/FlinkRelBuilder.scala

```diff
@@ -132,7 +134,23 @@ class FlinkRelBuilder(
       namedProperties: List[PlannerNamedWindowProperty],
       aggCalls: Iterable[AggCall]): RelBuilder = {
     // build logical aggregate
-    val aggregate = super.aggregate(groupKey, aggCalls).build().asInstanceOf[LogicalAggregate]
+
+    // Because of:
+    // [CALCITE-3763] RelBuilder.aggregate should prune unused fields from the input,
+    // if the input is a Project.
+    //
+    // the field can not be pruned if it is referenced by other expressions
+    // of the window aggregation(i.e. the TUMBLE_START/END).
+    // To solve this, we config the RelBuilder to forbidden this feature.
+    val aggregate = transform(
+      new UnaryOperator[RelBuilder.Config] {
+        override def apply(t: RelBuilder.Config)
+          : RelBuilder.Config = t.withPruneInputOfAggregate(false)
+      })
```

Comment:
I think we shouldn't treat this special here. The whole `FlinkRelBuilder` should be configured disabling this thing.

Reply:
For normal aggregation, there are many promotions in the plan, and i think we should keep the change.

The only case that need to disable this feature is the `WindowAggregation`, we generates the normal aggregation first then the window expressions.